tests/test_views_objects_permissions.py

```diff
@@ -183,3 +183,66 @@ def test_record_permissions_are_modified_by_patch(self):
                                    {'permissions': {'read': ['fxa:user']}},
                                    headers=self.headers)
         self.assertIn('fxa:user', resp.json['permissions']['read'])
+
+
+class ChildrenCreationTest(PermissionsTest):
+    def setUp(self):
+        self.app.put_json('/buckets/create',
+                          {'permissions': {'group:create': ['system.Authenticated']}},
+                          headers=self.alice_headers)
+        self.app.put_json('/buckets/write',
+                          {'permissions': {'write': ['system.Authenticated']}},
+                          headers=self.alice_headers)
+        self.app.put_json('/buckets/read',
+                          {'permissions': {'read': ['system.Authenticated']}},
+                          headers=self.alice_headers)
+        for parent in ('create', 'write', 'read'):
+            self.app.put_json('/buckets/%s/groups/child' % parent,
+                              MINIMALIST_GROUP,
+                              headers=self.alice_headers)
+
+    def test_cannot_read_others_objects_if_only_allowed_to_create(self):
+        self.app.get('/buckets/create/groups', headers=self.bob_headers, status=403)
+        self.app.get('/buckets/create/groups/child', headers=self.bob_headers, status=403)
+
+    def test_safe_creation_with_put_returns_412_if_allowed_to_create(self):
+        headers = self.bob_headers.copy()
+        headers.update({'If-None-Match': '*'})
```

Comment:
do you want to save those headers on the class so you can simply use `self.bob_if_none_match_headers`?